wikimedia/shiro-wordpress-theme
# #886: Modify search results to add sorting options buttons

## Changes
- Added sort dropdown with mobile responsiveness and positioning adjustments, follow other buttons and tabs design/colors
- Introduced `search-results-dropdown.js` to support dropdown functionality for search results, adding toggle functionality and making click outside of the dropdown to hide the dropdown
- Modified the search sorting mechanism to accept custom sort options from query parameters.
- It now supports ordering by relevance, date in both ascending and descending orders.

## Screenshot
![image](https://github.com/wikimedia/shiro-wordpress-theme/assets/90911997/d4f6d3a5-e150-4ce1-b050-477f6d5dcd62)

## Testing instructions
- Search for anything on the site
- Check the sort dropdown's appearance and responsiveness
- Test dropdown toggle functionality
- Verify sorting works by relevance, date (newest first), and date (oldest first)
- Ensure that external clicks hide the dropdown
